backend/de.metas.report/metasfresh-report-service/src/main/java/de/metas/report/jasper/OrgLogoLocalFileLoader.java

```diff
@@ -110,30 +116,81 @@ private File createAndGetLocalLogoFile(@NonNull final OrgId adOrgId)
 		return logoFile;
 	}
 
-	private I_AD_Image retrieveLogoImage(@NonNull final OrgId adOrgId)
+	private I_AD_Image retrieveImage(@NonNull final ResourceNameContext rsc)
 	{
+		final OrgId adOrgId = rsc.getOrgId();
 		if (adOrgId.isAny())
 		{
 			return null;
 		}
 
-		final Properties ctx = Env.getCtx();
+		if (rsc.isLogo())
+		{
+			return retrieveLogoImage(adOrgId);
+		}
+		else
+		{
+			return retrieveOtherImage(rsc);
+		}
+	}
+
+	private I_AD_Image retrieveOtherImage(@NonNull final ResourceNameContext rsc)
+	{
+		final OrgId adOrgId = rsc.getOrgId();
+		final String resourceName = rsc.getResourceName();
+
+		final List<String> listResourceName = Splitter.on('/')
```

Comment:
document what this part is doing.
consider extracting it to something like extractImageColumnName(context)

Reply:
consider adding some unit tests